### ISSUE TITLE
Revamp "infinite scrolling" addon

### ISSUE THREAD

**Issue by WorldLanguages** [ScratchAddons/ScratchAddons#5419]
### Why this would be helpful/solve a problem

We should revisit this old addon which has been made like 2 years ago. I think we could improve it a lot.

### What this change does

- Add infinite scrolling to more places, such as "What's happening" (if full-areas is enabled) and project search / explore (#2520)
- ~~Remove weird settings such as "fix footer", think of better ways to let users access the footer~~
- We removed support for studios at #3462

### Demonstration(s) and additional context

We need to be careful, because it's easy to add a bug that clicks "load more" infinitely, such as #2980

**Comment by Hans5958** [comment]
About the footer, it's just does not make sense to access the footer if what you want is an infinite stream of content. That setting is just doesn't make any sense to me.

**Comment by Hans5958** [comment]
The logic doesn't work here. 

Without a infinite scrolling feture, the reason of someone to scroll to the bottom is to access the footer, but with it, the reason of someone to scroll to the bottom is to have more stuff; more content.

- You don't scroll down a Scratch studio gallery or a YouTube playlist to access the footer. You want more projects from it.
- You don't scroll down a comments section (Scratch, YouTube) to access the footer. You want more comments of it.
- You don't scroll down the front page of YouTube, Twitter, Facebook, Reddit, TikTok, etc to access the footer. You want more content from it. This is why some sites, like YouTube or TikTok, put the content that should be on the footer, to the sidebar.

Even Never Ending Reddit, the "infinite scrolling" of [Reddit Enhancement Suite](https://redditenhancementsuite.com/) for old Reddit, doesn't offer a fixed footer, probably with the reason I mentioned.

If someone is often to access the footer, then they shouldn't using an infinite scrolling feature. A hybrid is nice, but it should put it where it is appropriate and nicely (TikTok and YouTube is appropriate), and I don't think there is a way to do it appropriately and/or nicely, including having a conspicuous fixed footer.

**Comment by RedGuy12** [comment]
related: https://github.com/ScratchAddons/ScratchAddons/issues/3071

**Comment by WorldLanguages** [comment]
> Remove weird settings such as "fix footer"

+1 from feedback

**Comment by WorldLanguages** [comment]
> Remove weird settings such as "fix footer", think of better ways to let users access the footer

Any ideas? @DNin01

**Comment by WorldLanguages** [comment]
Possibly we could do this:
- Remove the "fix footer" setting
- Keep the footer intact in pages where there's nothing to infinitely scroll in the main viewport (these pages will look as if the addon was disabled)
- As @Hans5958 suggested, do not provide any way to access the footer in pages where you can scroll infinitely. The user can go to the front page footer if necessary.

**Comment by BroJac5246** [comment]
> Possibly we could do this:
> - Remove the "fix footer" setting
> - Keep the footer intact in pages where there's nothing to infinitely scroll in the main viewport (these pages will look as if the addon was disabled)
> - As @Hans5958 suggested, do not provide any way to access the footer in pages where you can scroll infinitely. The user can go to the front page footer if necessary.

Yes, please! I never understood that setting. No one needs the footer that much.

**Comment by WorldLanguages** [comment]
New PR by me #6426 removes the setting, but it does not hide the footer if content is being dynamically added to the page.

**Comment by DNin01** [comment]
So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.

**Comment by BroJac5246** [comment]
> So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.

Because there isn't a good reason to use it.

**Comment by WorldLanguages** [comment]
> So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.

We can readd it later if people ask for it, maybe as a separate addon (or maybe not).

IMO the problem with that setting is that it is enabled by default and we get frequent feedback about it, so the actual fix for the problem would involve disabling this setting for everyone, including current users of the addon, as I would assume a very small portion of users intentionally keep this setting enabled.

**Comment by DNin01** [comment]
> > So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.
> 
> 
> 
> We can readd it later if people ask for it, maybe as a separate addon (or maybe not).
> 
> 
> 
> IMO the problem with that setting is that it is enabled by default and we get frequent feedback about it, so the actual fix for the problem would involve disabling this setting for everyone, including current users of the addon, as I would assume a very small portion of users intentionally keep this setting enabled.

So do people just want a better solution than the current "fix footer" feature?

**Comment by Samq64** [comment]
This was probably already mentioned somewhere, but if the setting is kept it should only do something where it makes sense.

**Comment by WorldLanguages** [comment]
> So do people just want a better solution than the current "fix footer" feature?

If you think about it, the "solution" is going to the front page to access the footer 🤔 
Showing the footer at all times in all pages makes no sense.

**Comment by BroJac5246** [comment]
> > So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.
> 
> Because there isn't a good reason to use it.

@DNin01 has your stance changed? If not then what is a good reason to keep and use it?

**Comment by DNin01** [comment]
> > > So wait, why is "fix footer" being removed? It seems like a good option to me, even if it isn't ideal.
> > 
> > 
> > Because there isn't a good reason to use it.
> 
> @DNin01 has your stance changed? If not then what is a good reason to keep and use it?

Whatever, I guess. I don't really have an opinion now. Everyone's made some good points.

**Comment by DNin01** [comment]
Closed by #6633, I believe?

**Comment by WorldLanguages** [comment]
I don't think so? Only 1 out of 3 items was about the fixed footer.

**Comment by DNin01** [comment]
> I don't think so? Only 1 out of 3 items was about the fixed footer.

Ah.
